File: p3_c/samples/TLoop.c
/** Test Loop: while, for, nested loops (1-D Array and 2-D Array)
 *  Program-ID:     TLoop.c
 *  Author:         Tyler Oalman
 *  OS:             Arch Linux
 *  Compiler:       GNU C
 */

#include <stdio.h>

void main ()
{
	int i, j;
	int a[3] = {1, 2, 3};
	int b[3][3] = {{10, 20, 30},
                       {40, 50, 60},
                       {70, 80, 90}};
	int *p;

	printf("\nTest while loop: 1-D Array and Pointer\n");
	p = &a[0];
	i = 0;
	while (i < 3)
	{
		printf("a[%d] = %d, *p = %d\t", i, a[i], *p);
		++p;
		++i;
	}

	printf("\nTest for loop: 2-D Array and Pointer\n");
	p = &b[1][0];
	for (j = 0; j < 3; j++)
	{
		printf("b[1,%d] = %d, *p = %d\t", j, b[1][j], *p);
		++p;
	}

	printf("\nTest nested loop: 2-D Array and Pointer\n");
	p = &b[0][0];
	for (i = 0; i < 3; i++)
	{
		printf("\n");
		for (j = 0; j < 3; j++)
		{
			printf("b[%d,%d] = %d, *p = %d\t", i, j, b[i][j], *p);
			++p;
		}
	}
	printf("\n");
}

File: p1_fortran/samples/TLoop.f
C Test Loops:   do-loop, if-goto-loop, nested loop
C Program-ID:   TLoop.f
C Author:       Tyler Oalman
C OS:           Arch Linux
C Compiler:     gcc-fortran

      PROGRAM TLoop
      INTEGER i, j, m, n
      INTEGER A(3, 4)
C Do-Loop:  The number of iterations is fixed at the beginning
C     DO label, loop-control-var = init-value, final-value, step-size
C         statements
C     label CONTINUE
      m = 1
      DO 10, i = 3, 6, 2
            m = m + i
10    CONTINUE
      WRITE (*,*) 'm = ', m
C if-goto-loop
      n = 1
      i = 3
20    IF (i .LE. 6) THEN
            n = n + i
            i = i + 2
            GOTO 20
      ENDIF
      WRITE (*,*) 'n = ', n
C nested Do-loop
      DO 40, i = 1, 3, 1
            DO 30, j = 1, 4, 1
                  A(i,j) = i + j
30          CONTINUE
40    CONTINUE
C nested if-goto-loop
      i = 1
50    IF (i .LE. 3) THEN
            j = 1
60          IF (j .LE. 4) THEN
                  WRITE (*,*) 'A (', i, ',', j, ') = ', A(i,j)
                  j = j + 1
                  GOTO 60
            ENDIF
            i = i + 1
            GOTO 50
      ENDIF
      END
C Input:    No
C Output:
C   m =            9
C   n =            9
C   A (           1 ,           1 ) =            2
C   A (           1 ,           2 ) =            3
C   A (           1 ,           3 ) =            4
C   A (           1 ,           4 ) =            5
C   A (           2 ,           1 ) =            3
C   A (           2 ,           2 ) =            4
C   A (           2 ,           3 ) =            5
C   A (           2 ,           4 ) =            6
C   A (           3 ,           1 ) =            4
C   A (           3 ,           2 ) =            5
C   A (           3 ,           3 ) =            6
C   A (           3 ,           4 ) =            7
